src/utils/pgwire/src/pg_server.rs

```diff
@@ -130,53 +128,99 @@ impl UserAuthenticator {
     }
 }
 
-/// Binds a Tcp listener at `addr`. Spawn a coroutine to serve every new connection.
+enum Listener {
+    Tcp(TcpListener),
+    Unix(UnixListener),
+}
+
+#[auto_enums::enum_derive(tokio1::AsyncRead, tokio1::AsyncWrite)]
+enum Stream {
+    Tcp(TcpStream),
+    Unix(UnixStream),
+}
+
+impl Listener {
+    async fn bind(addr: &str) -> io::Result<Self> {
+        if addr.starts_with("unix:") {
+            let path = &addr[5..];
```

Comment:
```suggestion
        if let Some(path) = addr.strip_prefix("unix:") {
```

Reply:
This is cool, but the cooler thing is that clippy can lint on this. 🤯